serverless/containers/how-to/add-a-custom-domain-to-a-container.mdx

```diff
@@ -36,11 +36,65 @@ Each container comes with an automatically generated domain, on which it is avai
+2. Ensure that the custom domain points to one of the Serverless Container existing endpoints: either the default one `....functions.fnc.<region>.scw.cloud`, or an already existing custom domain.
+3. If step 2 is OK, generate a dedicated TLS certificate for that custom domain so it can answer to HTTPS requests. Note: **during** that step, the custom domain might be available in plain HTTP, but it will be accessible in HTTPS only once this step is over.
+
+During the configuration of a custom domain (`pending` status), callers will likely receive HTTP 404 errors, while our gateways are reconfiguring themselves.
+
+The last 2 steps above are done simultaneously with a HTTP-01 challenge. If that challenge does not succeed after 3 minutes:
+
+- The custom domain will be put in `error` status.
+- The custom domain will not be reachable (in HTTP or HTTPS).
+- No TLS certificate will be generated.
+
+HTTP-01 challenge failure (and by extension, a custom domain in `error` status) can happen for the following reasons (outside of internal errors):
+
+| Issue | Description | How to fix or avoid this? |
+| ----- | ----------- | ----------- |
+| DNS record is not correctly configured. | If the DNS record (mostly CNAME) is misconfigured on your DNS provider side, and the custom domain to the Serverless Container default endpoint, we will not be able to configure the custom domain. | Ensure that on your DNS provider, a CNAME record links your custom domain to the Serverless Container. |
+| DNS record is not propagated yet. | This can be the case if the custom domain is created immediately after the CNAME is configured on your DNS provider side. | As a rule of thumb, wait a few minutes after creating the CNAME on your DNS provider, or at least wait until the DNS record is propagated on main resolvers (`1.1.1.1`, `8.8.8.8`). |
+| DNS cache is stale (still pointing to an old endpoint). | If the custom domain was pointing to another endpoint before adding the CNAME record to the Serverless Container, and if the TTL is greater than the maximum time of the check (3 min), it can sometimes happen that the custom domain still resolves to the old endpoint, thus making the challenge fail. | Wait until DNS is propagated and use a smaller TTL. |
+| Negative DNS cache is stale | If the initial check fails (DNS record is not propagated yet), and the negative TTL configured on your DNS provider side is high, the negative TTL will prevent subsequent checks to query the nameserver again, until it expires. Depending on the negative TTL configured, this can take more or less time. | Either reduce the negative TTL in your DNS provider configuration, or just wait until you are sure DNS record is propagated. If you already tried to add the custom domain and faced this issue, it is likely that you have to wait for the negative TTL to expire before making another attempt (so cache can also expire on our side). |
```

Comment:
nit: same, some wording simplification

```suggestion
| Negative DNS cache is stale | If the initial check fails (DNS record is not propagated yet), and the negative TTL configured on your DNS provider side is high, the negative TTL will prevent subsequent checks from querying the nameserver again, until it expires. Depending on the negative TTL configured, this can take more or less time. | Either reduce the negative TTL in your DNS provider configuration, or just wait until you are sure DNS record is propagated. If you already tried to add the custom domain and faced this issue, you likely have to wait for the negative TTL to expire before making another attempt (so the cache can also expire on our side). |
```